plugins/woocommerce/src/Admin/Schedulers/SchedulerTraits.php

```diff
@@ -256,9 +243,14 @@ public static function do_action_or_reschedule() {
 		// or schedule to run now if no blocking jobs exist.
 		$blocking_job = static::get_next_blocking_job( $action_name );
 		if ( $blocking_job ) {
-			$after = new \DateTime();
+			$next_action_time = self::get_next_action_time( $blocking_job );
+
+			if ( ! is_a( $next_action_time, 'DateTime' ) || $next_action_time < new \DateTime() ) {
```

Comment:
From my testing, it should also work without the `$next_action_time < new \DateTime()` condition, but I may be missing something. I would like to clarify why this condition is needed.

Reply:
I think it works without the condition. I added it to prevent actions from being scheduled into the past, which could be unexpected. I can remove the condition if necessary

Reply:
If I understand correctly, `self::get_next_action_time` should not return a time far beyond the present, and even if it is scheduled for a past time, it should still be able to function properly?

Going back to the fact that the main cause of the issue is that `self::get_next_action_time` may return `null`, then perhaps removing the second condition would improve the readability of this logic, and avoid the need to further explore the existence of the problem that the second condition is intended to deal with in the future.

Reply:
@eason9487 Yes, I agree. And the second condition added extra functionality that is not needed in this PR, so I removed it.